### ISSUE TITLE
Add documentation for Indexable#zip methods

### ISSUE THREAD

**Issue by rodrigopinto** [crystal-lang/crystal#5732]
Searching for the documentation about a `zip` method to be used over _Iterables_,  I realized the implementation of it is on the `Indexable.cr` and it is missing. 

- [ ] Add documentation to `Indexable#zip(other : Indexable)` 
- [ ] Add documentation to `Indexable#zip(other : Indexable) &block` 
- [ ] Add documentation to `Indexable#zip?(other : Indexable)` 
- [ ] Add documentation to `Indexable#zip?(other : Indexable) &block`

**Comment by RX14** [comment]
We should be able to implement `Enumerable#zip(other : Iterable)`. `Enumerable#zip(other : Enumerable)` is impossible though. I'm not sure how useful those methods would be though, although it feels "right" to me to add them.

**Comment by rodrigopinto** [comment]
@RX14 Yep, your point also makes sense, but I would treat as another issue.  It makes sense to be on `Enumerable` and that was the place were I did search first. However, I found them on [indexable.cr](https://github.com/crystal-lang/crystal/blob/master/src/indexable.cr#L415-L437) class and they are implemented already. 

What I am proposing in this issue is to add the documentation. I even already did it on [my branch](https://github.com/crystal-lang/crystal/compare/master...rodrigopinto:rp-add-docs-to-indexable-zip-methods) and I opened the issue to be proper evaluated.

WDYT?
